### DISCUSSION TITLE
ovos_bus_client API not documented

### DISCUSSION THREAD

**Discussion by geetee24** [OpenVoiceOS/ovos-buildroot#237] [Q&A] [answered]
mycroft works: client.emit( Message( 'recognizer_loop:utterance', { 'utterances': [ sys.argv [ 1 ] ], 'lang$

what is the API to send a message using ovos_bus_client please

Looked at ovos-bus-client and tried this.
......
Connecting to mycroft instance at 150.233.25.41
Sending message: hi there
2024-03-02 20:39:46.739 - OVOS - ovos_bus_client.client.client:on_error:118 - WARNING - Connection Refused. Is Messagebus Service running?

what does error mean?

**Comment by JarbasAl** [comment, marked as answer]
works the same way as mycroft

the error you are running into is because OVOS messagebus is not supposed to be accessed from external devices, this is a security measure, you seem to be trying to connect from a different machine given the ip address you posted, this is not safe and discouraged by default

more info here: https://openvoiceos.github.io/ovos-technical-manual/bus_service/

**Reply by geetee24** [reply]
the link to please check the companion project [HiveMind](https://openvoiceos.github.io/community-docs/friends/#hivemind)  is a 404.

please advise how to allow the bus to be controlled from anywhere in the same lan as one could do in MYcroft.   This is a big compatibility issue

**Reply by JarbasAl** [reply]
need to update that link for hivemind docs - https://jarbashivemind.github.io/HiveMind-community-docs/

this is not a compatibility issue, it is a security issue, you can make your OVOS as insecure as mycroft was by changing your websocket config "host" from 127.0.0.1 to 0.0.0.0. emphasis on the **insecure** part
